### ISSUE TITLE
Feature suggestion: Show TOC on blog index page

### ISSUE THREAD

**Issue by jeremy-feng** [squidfunk/mkdocs-material#4879]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... the documentation does not mention anything about my idea
- [X] ... there are no open or closed issues that are related to my idea

### Description

On the blog index page and archive pages, there is a blank area on the right side. I suggest to enable TOC there, like on the category page.

### Use Cases

With TOC, we can easily find what contents are in this page. Now the right side shows nothing, so we can use it to show TOC.

### Screenshots / Mockups

On the index page and archive pages, the right side shows nothing.

![image](https://user-images.githubusercontent.com/44312563/212876440-56aa67da-8489-4b28-934b-a76201785aeb.png)

On category pagewe can see a TOC on the right side.

![image](https://user-images.githubusercontent.com/44312563/212876854-e6b7b4a9-f57a-4f46-860f-7fdc9591a0a9.png)

**Comment by squidfunk** [comment]
Added support for index page and archive page table of contents in [`b0f38f0`](https://github.com/squidfunk/mkdocs-material-insiders/commit/b0f38f073a80cba04f2faae43acf627bba1390be). Additionally, the settings were generalized. You can now switch on table of contents for the entire blog with a single setting, and override settings if necessary:

``` yaml
plugins:
  - blog:
      blog_toc: true       # enable for blog, archive and category index pages
      archive_toc: true    # enable for archive index pages
      categories_toc: true # enable for category index pages
```

**Comment by squidfunk** [comment]
Added documentation in 921a6302a.

**Comment by squidfunk** [comment]
Released as part of [9.0.13+insiders-4.31.0](https://github.com/squidfunk/mkdocs-material-insiders/releases/tag/9.0.13-insiders-4.31.0).
